### PR TITLE
Added security role and configuration

### PR DESCRIPTION
**Pull request by YPCrumble**
@jcalazan here's an initial PR for #56 - would be great to hear what you think!

Currently the security role is completely separate and run once as an initial configuration step via security.yml. I tried a different way of implementing the Security role via an explicit flag per your suggestion. The problem is that the initial login uses the root username and subsequent logins use whatever username you specify in the Security role. Ansible doesn't really support dynamic variables - nor should it - so unless you have another idea for me this was the best option.

I've tested this on Ubuntu 14.04 but just the ansible script - once you're OK with how this looks I'll test on the Vagrant machine.

### CODE REVIEW

**Comment by YPCrumble** [comment]
@jcalazan a related question is whether you'd be interested in [adding newlines](http://stackoverflow.com/questions/729692/why-should-text-files-end-with-a-newline) at the end of files? I just realized the conflicts in my first PR that I deleted were due to these.

**Comment by jcalazan** [comment]
@YPCrumble  Sorry, haven't had a chance to take a look, been a bit busy.  Will hopefully get to it this weekend.

**Comment by YPCrumble** [comment]
@jcalazan no problem, thank you!

**Comment by jcalazan** [comment]
Hey @YPCrumble ,

Sorry for the delay, I took a quick look and looks good so far :).  Just a few comments.

- Can we rename `server_user_name` to just `server_user` (for consistency with Ansible's naming) and move the var in `roles/base/defaults/main.yml` with default value of `root`?  The reason being is not everyone may want to use the custom user/security role and in the case a user wants to use different values per host then they can override it in the inventory file.

- Can we move the vars `server_user_password` and `run_aptitude_safe_upgrade` to `roles/security/defaults/main.yml`? The user may want different values for these as well per host and they'd be able to do it then from the inventory file.

- For unattended upgrades, UFW, and Fail2Ban, can we add vars so users can select only those security features they want? (for example: `enable_unattended_upgrades: yes`, `enable_ufw: yes`, etc.).

Let me know if you see issues with the above and thanks for doing this!

**Comment by jcalazan** [comment]
Oh, for the newline at end of files, let's create a separate PR for it once this one is merged. :)

**Comment by YPCrumble** [comment]
@jcalazan thanks! All changes make sense - will make an update soon. I like the toggles on the unattended-upgrades, UFW, and fail2ban - I'll probably set the first two to `yes` by default and fail2ban to `no` as that's the one that's most likely to lock someone out.

File: roles/security/tasks/main.yml [hunk 1]
```diff
@@ -22,35 +23,51 @@
               regexp="{{ server_user }} ALL"
               line="{{ server_user }} ALL=(ALL) ALL"
               state=present
+  when: server_user != "root"
 
-- name: Install required packages
-  apt: update_cache={{ update_apt_cache }} state=installed pkg={{ item }}
-  with_items: 
-  - "{{ required_packages }}"
+# Unattended Upgrades settings
```
Comment (YPCrumble) [inline roles/security/tasks/main.yml]: @jcalazan I wasn't sure of a way to continue looping over the `required_packages` variable while also making each package optional. Therefore I am installing each packages separately in the `tasks` section. If you have a better solution let me know?

**Comment by YPCrumble** [comment]
@jcalazan Updated and tested on Ubuntu 14.04. Let me know any further suggestions? Thanks for your review!

File: roles/security/tasks/main.yml [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+  apt: name=sudo update_cache={{ update_apt_cache }} state=present cache_valid_time=86400
+
+- name: add user to sudoers
+  lineinfile: dest=/etc/sudoers
+              regexp="{{ server_user }} ALL"
+              line="{{ server_user }} ALL=(ALL) ALL"
+              state=present
+  when: server_user != "root"
+
+# Unattended Upgrades settings
```
Comment (jcalazan) [inline roles/security/tasks/main.yml]: Just to keep things a bit more organized, can we put these security features into separate yml files.  For example, `setup_unattended_upgrades.yml`.  Then in `main.yml` we can do an include with a `when:` condition for that include.  This way we'll also only need to use the `when:` condition once per security feature.

File: roles/security/tasks/main.yml [hunk 3]
```diff
@@ -0,0 +1,103 @@
+---
+- name: Update APT packages cache
+  apt: update_cache={{ update_apt_cache }} cache_valid_time=86400
+
+- name: Perform aptitude safe-upgrade
+  apt: upgrade=yes
+  when: run_aptitude_safe_upgrade is defined and run_aptitude_safe_upgrade
+
+- name: add user
```
Comment (jcalazan) [inline roles/security/tasks/main.yml]: Can we capitalize the first letter for consistency?

**Comment by jcalazan** [comment]
@YPCrumble This looks good! The only other thing I'd change is separating the security features into separate yml files and doing an include from `main.yml`.

Also, just for consistency, I'd like the task names cap first (we can leave the handlers names all lowercase).

Example:
 
````
# main.yml

- include: enable_ufw.yml
  tags: security.ufw
  when: enable_ufw is defined and enable_ufw
````

Thanks man!

File: security.yml [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+  remote_user: root
+  vars:
+    - update_apt_cache: yes
+  vars_files:
+    - env_vars/base.yml
+    - env_vars/{{ env }}.yml
+    - roles/base/defaults/main.yml
+
+  roles:
+    - { role: security, tags: ['security'] }
```
Comment (jcalazan) [inline security.yml]: Is the `tags: ['security']` part here necessary?

**Comment by YPCrumble** [comment]
@jcalazan just pushed the updates you requested and tested on Ubuntu 14.04. Let me know if any issues? Thanks again for the great code reviews!

**Pull request merged**